pandas/core/internals/blocks.py

```diff
@@ -2924,6 +2924,30 @@ def where(
             )
         return result
 
+    def replace(
+        self,
+        to_replace,
+        value,
+        inplace: bool = False,
+        filter=None,
+        regex: bool = False,
+        convert: bool = True,
+    ):
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        result = self if inplace else self.copy()
+        if filter is None:
+            result.values.replace(to_replace, value, inplace=True)
+            if convert:
+                return result.convert(numeric=False, copy=not inplace)
+            else:
+                return result
+        else:
+            if not isna(value):
+                result.values.add_categories(value, inplace=True)
+            return super(CategoricalBlock, result).replace(
+                to_replace, value, inplace, filter, regex, convert
+            )
```

Comment:
can any of the logic in this method generalize to ExtensionBlock?  (presumably we'd need to add `replace` to the EA interface?)

Reply:
this is a good point, but we can certainly do as a followup; can you open an issue for discussion.